File: apps/sdkwork-cloudrouter-flutter-mobile/lib/bootstrap/runtime.dart
import 'package:sdkwork_cloudrouter_flutter_mobile_core/sdkwork_cloudrouter_flutter_mobile_core.dart';

import 'host_adapters.dart';
import 'iam_runtime.dart';
import 'routes.dart';
import 'sdk_clients.dart';

/// Root runtime assembled by `bootstrap()` in the order required by
/// `APP_SDK_INTEGRATION_SPEC.md` section 4: environment, token/identity, host
/// adapters, generated app SDK clients, then routes.
class CloudRouterFlutterRuntime {
  const CloudRouterFlutterRuntime({
    required this.sdkClients,
    required this.consolePorts,
    required this.session,
  });

  final CloudRouterAppSdkClients sdkClients;
  final CloudRouterConsolePorts consolePorts;
  final CloudRouterSession session;
}

Future<CloudRouterFlutterRuntime> bootstrap() async {
  final sdkClients = createSdkClients();
  final session = readCloudRouterSession();
  createIamRuntime(session);
  registerHostAdapters();
  final consolePorts = createCloudRouterConsolePorts(sdkClients: sdkClients);
  createRoutes();
  return CloudRouterFlutterRuntime(
    sdkClients: sdkClients,
    consolePorts: consolePorts,
    session: session,
  );
}

File: apps/sdkwork-cloudrouter-h5/src/bootstrap/runtime.ts
import {
  loadRuntimeEnv,
  resolveCloudRouterAppApiBaseUrl,
} from '@sdkwork/cloudrouter-h5-core/sdk';

import { createHostEnvironment } from './environment.js';
import { registerHostAdapters } from './hostAdapters.js';
import { bootstrapIamRuntime } from './iamRuntime.js';
import { bootstrapH5SdkClients } from './sdkClients.js';
import { configureConsoleRuntime } from '../providers/AppProviders.js';

/**
 * Single application bootstrap: environment selection, host adapters, the IAM
 * runtime, the generated app SDK client, and the console ports, in the order
 * required by `APP_SDK_INTEGRATION_SPEC.md` §4.
 *
 * BROWSER_RUNTIME_ENV_SPEC.md §3/§4: `/runtime-env.json` is the deploy-time
 * authority for profile identity and SDK bases. It is loaded — which also
 * publishes the `SDKWORK_RUNTIME_ENV` bridge — before any SDK client is
 * constructed, so a deployed artifact never falls back to build-time defaults.
 */
export async function bootstrapH5Application(): Promise<void> {
  const runtimeEnv = await loadRuntimeEnv();
  const environment = createHostEnvironment(
    import.meta.env as unknown as Record<string, unknown>,
    resolveCloudRouterAppApiBaseUrl(runtimeEnv),
  );
  registerHostAdapters();
  bootstrapIamRuntime();
  const sdk = bootstrapH5SdkClients();
  configureConsoleRuntime({
    platform: 'h5',
    environment: environment.lifecycleEnvironment,
    profileId: environment.profileId,
    ports: sdk.consolePorts,
  });
}
